### DISCUSSION TITLE
QQ: Does the lamba deployed respect the timeout set in template?

### DISCUSSION THREAD

**Discussion by JohnPreston** [aws-cloudformation/cloudformation-cli-python-plugin#196] [Q&A]
Hello,

working on a resource at the moment which cannot make use of the "In progress" status, as it has to complete the job it is doing synchronously (in the current implementation anyway). Given there is a template that defines the lambda function "properties" I wanted to make sure that the timeout property is respected.

And if not, what's the default timeout of the "backend lambda" that manages the resources ?

Thank you,

**Comment by ericzbeard** [comment]
Can you provide a bit more information on this one?

**Reply by JohnPreston** [reply]
When the resource is enabled in a customer account, my assumption is that the code of the resource is executed with AWS Lambda. Now, if that is the case, then that lambda has an execution time before being cut off, with a maximum of 15 minutes.

The use of lambda is my assumption given how the automated tests run with SAM local and it generally makes sense, considering that the reply to CFN can be "IN_PROGRESS", and the function would be called upon multiple times, until the status and return changes.

Here in this use-case, the resource creation can be done every time within a reasonable amount of time, under 2 minutes. But it needs at least that much, just to be sure.

So when creating a resource, how can we influence the timeout of the function executing the code? Whether CFN resource use lambda or not to do the job.

Hope that makes sense?
